src/osd/PG.cc

```diff
@@ -2064,26 +2064,29 @@ void PG::mark_clean()
 unsigned PG::get_recovery_priority()
 {
   // a higher value -> a higher priority
-  return OSD_RECOVERY_PRIORITY_MAX;
+  unsigned ret = OSD_RECOVERY_PRIORITY_BASE + pool.info.recovery_priority;
+  if (ret > OSD_RECOVERY_PRIORITY_MAX)
+    ret = OSD_RECOVERY_PRIORITY_MAX;
+  return ret;
 }
 
 unsigned PG::get_backfill_priority()
 {
   // a higher value -> a higher priority
-
-  // undersized: 200 + num missing replicas
+  unsigned ret = OSD_BACKFILL_PRIORITY_BASE;
   if (is_undersized()) {
+    // undersized: OSD_BACKFILL_DEGRADED_PRIORITY_BASE + num missing replicas
     assert(pool.info.size > actingset.size());
-    return 200 + (pool.info.size - actingset.size());
-  }
+    ret = OSD_BACKFILL_DEGRADED_PRIORITY_BASE + (pool.info.size - actingset.size());
+  } else if (is_degraded()) {
+    // degraded: baseline degraded
+    ret = OSD_BACKFILL_DEGRADED_PRIORITY_BASE;
+  } 
 
-  // degraded: baseline degraded
-  if (is_degraded()) {
-    return 200;
-  }
+  if (ret > OSD_RECOVERY_PRIORITY_MAX)
+    ret = OSD_RECOVERY_PRIORITY_MAX;
```

Comment:
I assume it would be bad for backfill priority to exceed OSD_RECOVERY_PRIORITY_MAX.  This is unlikely since pool.info.size would have to be greater than (OSD_RECOVERY_PRIORITY_MAX -  OSD_BACKFILL_DEGRADED_PRIORITY_BASE + actingset.size()) which is over 105.